src/include/txn.i

```diff
@@ -791,12 +805,25 @@ __wt_txn_read_upd_list(WT_SESSION_IMPL *session, WT_UPDATE *upd, WT_UPDATE **upd
               F_ISSET(session, WT_SESSION_IGNORE_HS_TOMBSTONE) &&
               (upd->start_ts != WT_TS_NONE || upd->txnid != WT_TXN_NONE))
                 continue;
-            *updp = upd;
-            return (0);
+            break;
         }
         if (upd_visible == WT_VISIBLE_PREPARE)
             return (WT_PREPARE_CONFLICT);
     }
+    if (upd == NULL)
+        return (0);
+    /*
+     * Now assign to the update value. If it's not a modify, we're free to simply point the value at
+     * the update's memory without owning it. If it is a modify, we need to reconstruct the full
+     * update now and make the value own the buffer.
+     *
+     * If the caller has specifically asked us to skip assigning the buffer, we shouldn't bother
+     * reconstructing the modify.
+     */
+    if (upd->type != WT_UPDATE_MODIFY || cbt->upd_value->skip_buf)
```

Comment:
Should we just do nothing is we want to skip_buf?

Reply:
Sorry, I get it.

Reply:
No, we want to assign everything else except the buffer which is handled inside `__wt_upd_value_assign`.